Add two tests that document portability issues of run-records

mih opened the pull request:
- IO specifications are stored in platform-native conventions
- IO specifications are reported as-is

combined with the absence of any platform type record, this makes run-records non-portable across unix/windows scopes.

Ping https://github.com/datalad/datalad/issues/7512

TODO

- [ ] rethink the interpretation of old run-records. I tend to believe that we should never guess. We should add a config switch how run-records are to be intepreted. By default, all recorded paths should be considered POSIX. By config, we should be able to switch that to `windows` (not to `native`). This could/would be considered for reading (`rerun`) AND writing (`run`). If incoming paths are incompatible with the setting (e.g., absolute windows path and a POSIX setting), an error would be issued.

mih committed:
Add two tests that document portability issues of run-records

- IO specifications are stored in platform-native conventions
- IO specifications are reported as-is

combined with the absence of any platform type record, this makes
run-records non-portable across unix/windows scopes.

Ping https://github.com/datalad/datalad/issues/7512

adswa commented on datalad_next/patches/tests/test_run.py:
```diff
@@ -23,3 +26,79 @@ def test_substitution_config_default(existing_dataset):
+        '{python} -c "'
+        'from shutil import copyfile;'
+        'from pathlib import Path;'
+        r'[copyfile(f, Path.cwd() / \'outputs\' / Path(f).name)'
```
```suggestion
        r'[copyfile(f, Path.cwd() / \"outputs\" / Path(f).name)'
```

There is one current test failure because of this
```
[INFO] == Command start (output follows) ===== 
  File "<string>", line 1
    from shutil import copyfile;from pathlib import Path;[copyfile(f, Path.cwd() / \'outputs\' / Path(f).name)  for f in "inputs/testfile.txt".split()]
                                                                                                                                                      ^
SyntaxError: unexpected character after line continuation character
[INFO] == Command exit (modification check follows) ===== 
FAILED
```

mih replied:
I actually changed it from the alternative you propose -- but that fails on windows due to an unfortunate quoting behavior

mih committed:
Store run-record path specs POSIX and report platform paths

There is limited support for reading and acting on old run-records
that have paths stored in platform conventions. Detection works
when the path matching an existing item on the file system.

Paths are always stored in POSIX notation, whenever they are relative.
